### ISSUE TITLE
update container annotations

### ISSUE THREAD

**Issue by jku** [theupdateframework/tuf#2722]
Once #2721 is merged, we could require python 3.8 and use the nicer annotations and features. Known improvements:
* annotations for standard collections: `dict` instead of `typing.Dict`
* dict unions (I believe we have at least one place where this would be useful)

Vague process:
* bump requires-python in pyproject.toml
* run `ruff check --fix --unsafe-fixes tuf tests examples`
* review that changes made by ruff are sensible
* check if RootVerificationResult `signed()` and `unsigned()` work with a dict union -- this step could be left to another PR
* make sure `tox -e lint` and `tox -e test` pass

**Comment by jku** [comment]
> * `str | None` instead of `Optional[str]`

Actually this might be in 3.10 only -- if that is true then we don't want to change it yet
